homeassistant/components/mqtt/scene.py

```diff
@@ -86,47 +85,22 @@ class MqttScene(
 
     def __init__(self, hass, config, config_entry, discovery_data):
         """Initialize the MQTT scene."""
-        self.hass = hass
-        self._state = False
-        self._sub_state = None
-
-        self._unique_id = config.get(CONF_UNIQUE_ID)
-
-        # Load config
-        self._setup_from_config(config)
+        MqttEntity.__init__(self, hass, config, config_entry, discovery_data)
 
-        # Initialize entity_id from config
-        self._init_entity_id()
-
-        MqttAvailability.__init__(self, config)
-        MqttDiscoveryUpdate.__init__(self, discovery_data, self.discovery_update)
-
-    def _init_entity_id(self):
-        """Set entity_id from object_id if defined in config."""
-        init_entity_id_from_config(
-            self.hass, self, self._config, self._entity_id_format
-        )
-
-    async def async_added_to_hass(self):
-        """Subscribe to MQTT events."""
-        await super().async_added_to_hass()
-        self.async_send_discovery_done()
-
-    async def discovery_update(self, discovery_payload):
-        """Handle updated discovery message."""
-        config = DISCOVERY_SCHEMA(discovery_payload)
-        self._setup_from_config(config)
-        await self.availability_discovery_update(config)
-        self.async_write_ha_state()
+    @staticmethod
+    def config_schema():
+        """Return the config schema."""
+        return DISCOVERY_SCHEMA
 
     def _setup_from_config(self, config):
         """(Re)Setup the entity."""
         self._config = config
 
-    async def async_will_remove_from_hass(self):
-        """Unsubscribe when removed."""
-        await MqttAvailability.async_will_remove_from_hass(self)
-        await MqttDiscoveryUpdate.async_will_remove_from_hass(self)
+    def _prepare_subscribe_topics(self):
+        """(Re)Subscribe to topics."""
+
+    async def _subscribe_topics(self):
+        """(Re)Subscribe to topics."""
 
     @property
     def name(self):
```

Comment:
We can remove this, as well as the unique_id and icon properties

Reply:
Done